syntax = "proto3";

package google.protobuf.compiler.wasm.util;

message GeneratorInput {
  string name = 1;
  string content = 2;
  string language = 3;
  string parameters = 4;
}

message GeneratorOutput {
  string error = 1;
  map<string, string> files = 2;
}

message LanguageDescription {
  string protoc_output_flag = 1;
  string protoc_option_flag = 2;
}

message LanguageList {
  map<string, LanguageDescription> languages = 1;
}
